src/toil/jobStores/awsJobStore.py

```diff
@@ -163,13 +166,20 @@ def getPublicUrl(self, jobStoreFileID):
         """
         For Amazon SimpleDB requests, use HTTP GET requests that are URLs with query strings.
         http://awsdocs.s3.amazonaws.com/SDB/latest/sdb-dg.pdf
-        Create url, check if valid, return.
         Encrypted file urls are currently not supported
         """
         key = self.files.get_key(key_name=jobStoreFileID)
+        if key is None:
```

Comment:
This function has always been broken. We should get the version from SDB first and then get that individual version. Otherwise we risk handing out a URL to an old version.